js/src/forum/components/Poll/PollOption.tsx

```diff
@@ -50,24 +45,39 @@ export default class PollOption extends Component<PollOptionAttrs, PollState> {
   }
 
   view(): Mithril.Children {
+    // following values can be changed by ui interactions, so we need to update them on every render
+    this.hasVoted = this.state.hasVoted();
+    this.totalVotes = this.state.overallVoteCount();
+    this.votes = this.option.voteCount();
     this.voted = this.state.hasVotedFor(this.option);
 
     const isDisabled = this.state.loadingOptions || (this.hasVoted && !this.poll.canChangeVote());
     const width = this.canSeeVoteCount ? this.percent() : (Number(this.voted) / (this.poll.myVotes()?.length || 1)) * 100;
 
     const bar = (
-      <div className="PollBar" data-selected={!!this.voted} style={`--poll-option-width: ${width}%`}>
+      <label className="PollBar" data-selected={!!this.voted} style={`--poll-option-width: ${width}%`}>
         {this.state.showCheckMarks && (
-          <label className="PollAnswer-checkbox checkbox">
-            <input onchange={this.state.changeVote.bind(this.state, this.option)} type="checkbox" checked={this.voted} disabled={isDisabled} />
+          <div className="PollAnswer-checkbox">
+            <input
+              className="PollAnswer-input sr-only"
+              type="checkbox"
+              id={this.option.id()}
+              name={this.name}
+              value={this.answer}
+              checked={this.voted}
+              disabled={isDisabled}
+              aria-labelledby={`${this.name}-${this.option.id()}-label`}
```

Comment:
add the related id on the "label" element -> .PollAnswer-text-answer

Reply:
What is the related id?